src/App.jsx

```diff
@@ -1,53 +1,64 @@
 import { useState } from 'react';
+import { Routes, Route, Outlet, Navigate } from 'react-router-dom';
 
-import Container from './common/Container/Container';
+import './App.css';
 
 import { mockedCoursesList, mockedAuthorsList } from './constants';
 
-import './App.css';
-
 import Header from './components/Header/Header';
+import Container from './common/Container/Container';
 import Courses from './components/Courses/Courses';
 import CourseInfo from './components/CourseInfo/CourseInfo';
-import EmptyCourseList from './components/EmptyCourseList/EmptyCourseList';
-
-function App() {
-	const [selectedCourse, setSelectedCourse] = useState(null);
-
-	const handleCourseInfo = (course) => {
-		setSelectedCourse(course);
-	};
-
-	const renderContent = () => {
-		if (selectedCourse) {
-			return (
-				<CourseInfo
-					{...selectedCourse}
-					authorsList={mockedAuthorsList}
-					onBackToCourses={handleCourseInfo}
-				/>
-			);
-		} else if (mockedCoursesList.length) {
-			return (
-				<Courses
-					coursesList={mockedCoursesList}
-					authorsList={mockedAuthorsList}
-					onSelectCourse={handleCourseInfo}
-				/>
-			);
-		} else {
-			return <EmptyCourseList />;
-		}
-	};
+import CreateCourse from './components/CreateCourse/CreateCourse';
+import Registration from './components/Registration/Registration';
+import Login from './components/Login/Login';
+import Footer from './components/Footer/Footer';
+
+const token = localStorage.getItem('token');
 
+const Layout = () => {
 	return (
 		<>
 			<Header />
 			<main>
-				<Container>{renderContent()}</Container>
+				<Container>
+					<Outlet />
+				</Container>
 			</main>
+			<Footer />
 		</>
 	);
-}
+};
+
+const App = () => {
+	const [courses, setCourses] = useState(mockedCoursesList);
+
+	return (
+		<Routes>
+			<Route path='/' element={<Layout />}>
+				<Route
+					path='courses'
+					element={
+						<Courses coursesList={courses} authorsList={mockedAuthorsList} />
+					}
+				/>
+				<Route
+					path='courses/:courseId'
+					element={
+						<CourseInfo coursesList={courses} authorsList={mockedAuthorsList} />
+					}
+				/>
+				<Route
+					path='courses/add'
+					element={<CreateCourse courses={courses} setCourses={setCourses} />}
+				/>
+				<Route path='register' element={<Registration />} />
```

Comment:
```suggestion
				<Route path='registration' element={<Registration />} />
```
register is the endpoint on the server, the route should be registration
https://d17btkcdsmqrmh.cloudfront.net/react-fundamentals/docs/module-2/home-task/task#registration-component

please accept this suggestion